client/search-ui/src/input/SearchHistoryDropdown.tsx

```diff
@@ -0,0 +1,179 @@
+    KeyboardEventHandler,
+    useRef,
+    MouseEventHandler,
+    MouseEvent,
+} from 'react'
+
+import { mdiClockOutline } from '@mdi/js'
+import classNames from 'classnames'
+
+import { SyntaxHighlightedSearchQuery } from '@sourcegraph/search-ui'
+import { shortcutDisplayName } from '@sourcegraph/shared/src/keyboardShortcuts'
+import { Shortcut } from '@sourcegraph/shared/src/react-shortcuts'
+import { RecentSearch } from '@sourcegraph/shared/src/settings/temporary/recentSearches'
+import { TelemetryProps } from '@sourcegraph/shared/src/telemetry/telemetryService'
+// eslint-disable-next-line no-restricted-imports
+import { Timestamp } from '@sourcegraph/web/src/components/time/Timestamp'
+import {
+    Icon,
+    Popover,
+    PopoverTrigger,
+    PopoverContent,
+    usePopoverContext,
+    Flipping,
+    Tooltip,
+    PopoverOpenEvent,
+    createRectangle,
+} from '@sourcegraph/wildcard'
+
+import styles from './SearchHistoryDropdown.module.scss'
+
+const buttonContent: React.ReactElement = <Icon svgPath={mdiClockOutline} aria-hidden="true" />
+
+interface SearchHistoryDropdownProps extends TelemetryProps {
+    className?: string
+    recentSearches: RecentSearch[]
+    onSelect: (search: RecentSearch) => void
+}
+
+const recentSearchTooltipContent = `Recent searches ${shortcutDisplayName('Mod+ArrowDown')}`
+// Adds padding to the popover content to add some space between the trigger
+// button and the content
+const popoverPadding = createRectangle(0, 0, 0, 2)
+
+export const SearchHistoryDropdown: React.FunctionComponent<SearchHistoryDropdownProps> = React.memo(
+    ({ recentSearches = [], onSelect, className, telemetryService }) => {
+        const [isOpen, setIsOpen] = useState(false)
+
+        const handlePopoverToggle = useCallback(
+            (event: PopoverOpenEvent): void => {
+                setIsOpen(event.isOpen)
+                telemetryService.log(event.isOpen ? 'RecentSearchesListOpened' : 'RecentSearchesListDismissed')
+            },
+            [telemetryService, setIsOpen]
+        )
+
+        const openOnShortcut = useCallback(() => {
+            telemetryService.log('RecentSearchesListOpened')
+            setIsOpen(true)
+        }, [telemetryService, setIsOpen])
+
+        const onSelectInternal = useCallback(
+            (search: RecentSearch) => {
+                telemetryService.log('RecentSearchSelected')
+                onSelect(search)
+                setIsOpen(false)
+            },
+            [telemetryService, onSelect, setIsOpen]
+        )
+
+        return (
+            <>
+                <Popover isOpen={isOpen} onOpenChange={handlePopoverToggle}>
+                    <Tooltip content={recentSearchTooltipContent}>
+                        <PopoverTrigger
+                            type="button"
+                            className={classNames(styles.triggerButton, isOpen ? styles.open : null, className)}
+                            aria-label="Toggle search history"
```

Comment:
Focused is locked inside when open, so can only toggle to open.
```suggestion
                            aria-label="Open search history"
```